test/unix/io_array.ml

```diff
@@ -18,10 +18,12 @@ module Entry = struct
     (key, value)
 
   let append_io io (key, value) =
-    let encoded_key = Key.encode key in
-    let encoded_value = Value.encode value in
-    IO.append io encoded_key;
-    IO.append io encoded_value
+    Key.encode key (IO.append ~auto:false io);
+    Value.encode value (IO.append ~auto:false io);
+    (* Even though the key is an unary value, if a crash occurs between the
+       [Key.encode] and [Value.encode] calls, the flushed value will be
+       incomplete. *)
+    IO.flush io
```

Comment:
I suspect this is going to trigger *a lot* of flushes, presumably you need something that flushes only when necessary here?

Reply:
Or convert `append` to a batch-compatible form too:

```ocaml
  IO.append_transaction io (fun f ->
    Key.encode key f;
    Value.encode value f)
```

... where the IO can take care of auto-flushing only after the transaction is over. So something like:

```ocaml
  type consumer := (string -> unit) -> unit
  val append_batch : io -> (consumer -> 'a) -> 'a
```

Reply:
Thanks for the advice! The last update should look like what you said :-)